blackbird_python/blackbird/tests/test_program.py

```diff
@@ -403,3 +405,90 @@ def test_serialize_operation_multiple_arrays(self):
             """
         )
         assert res == expected
+
+
+class TestProgramIntegration:
+
+    def test_template_with_arrays(self):
+        """Test templates can be initialized with parameters inside of arrays"""
+        template=dedent(
+            """\
+            name template_tdm
+            version 1.0
+
+            float array one[1, 2] =
+                {p_one}
+
+            float array two[2, 3] =
+                1, {p_two}, 3
+                1, 2, {p_three}
+
+            int four = {p_four}
+
+            for int i in 0:2
+                Sgate(two[i], 0) | 1
+                BSgate(one[i]) | (1, 0)
+                MeasureHomodyne(four) | 0
+            """
+        )
+
+        bb = loads(template)
+
+        assert bb.parameters == {
+            'p_one_0_0', 'p_one_0_1', 'p_two', 'p_three', 'p_four'
+        }
+
+        assert bb.is_template()
+
+        bb2 = bb(p_one=[[11, 12]], p_two=22, p_three=33, p_four=44)
```

Comment:
Would `p_one` have to be passed as an array? Or is it restricted to `list`?

Reply:
It can be either a list or an array in this case. It's automagically cast as a NumPy array later on.

Reply:
Maybe worth parametrizing the test/passing a parameter as an array?